BatchApex/BatchApex.cls

```diff
@@ -0,0 +1,119 @@
+    }
+    
+    /**
+     * Wrapper class with opportunity and sum
+     */
+    public class WrapperClass {
+         List<Opportunity> opportunityList;
+         Double sum;
+         
+        /**
+         *Constructor to initialize the warpper class variables
+         */
+        public WrapperClass() {
+            opportunityList = new List<Opportunity>();
+            sum = 0;
+        }
+    }
+    
+    /**
+     * Get all opportunities whos lookup field to account is not blank 
+     * and email field on Account object is not blank
+     */
+    global Database.QueryLocator start(Database.BatchableContext batchableContext) {
+            return Database.getQueryLocator([
+                SELECT 
+                    Id, 
+                    Name, 
+                    Amount, 
+                    StageName, 
+                    Opportunity.Account.Email_Address__c, 
+                    AccountId 
+                FROM 
+                    Opportunity 
+                WHERE 
+                    AccountId <> null 
+                AND 
+                    Opportunity.Account.Email_Address__c <> null
+            ]);
+    }
+    
+    /**
+     * This will sum up all the closed won opportunities of a perticular account per day
+     * and will send a mail to the primary contact
+     */
+    global void execute(Database.BatchableContext batchableContext,
+         List<Opportunity> opportunityparameter) {
+        for(Opportunity opportunityInstance : opportunityparameter) {
+            if(accountOpportunityMap.containsKey(opportunityInstance.Account.Email_Address__c)) {
+                WrapperClass wrapperInstance = accountOpportunityMap.get(
+                    opportunityInstance.Account.Email_Address__c);
+                
+                // Add the opportunity in the opportunity list in the wrapper class
+                wrapperInstance.opportunityList.add(opportunityInstance);
+                
+                // Add the amount of that opportunity which is inserted in the opportunity list 
+                // in the sum variable in the wrapper class
+                wrapperInstance.sum += opportunityInstance.Amount;
+                accountOpportunityMap.put(
+                    opportunityInstance.Account.Email_Address__c, wrapperInstance);
+            }
+            else {
+                wrapperClass wrapperInstance = new wrapperClass();
+                wrapperInstance.opportunityList.add(opportunityInstance);
+                wrapperInstance.sum = opportunityInstance.Amount;
+                accountOpportunityMap.put
+                    (opportunityInstance.Account.Email_Address__c, wrapperInstance);
+            }
+        }
+    }
+     
+      global void execute(SchedulableContext schedulableContext) {
+          
+      }
+     
+     /**
+      * The finish block to send mail to primary contact of all opportunities 
+      * with the with their ammount
```

Comment:
Comment incomplete. What does " with the with their ammount" means? Also correct the amount spelling